src/ui/page/dataverse/dataspace/governance/governanceNavigation.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable max-lines-per-function */
+import type { FC } from 'react'
+import { useEffect, useCallback, useState } from 'react'
+import { NavLink } from 'react-router-dom'
+import classnames from 'classnames'
+import { useAppStore } from '@/ui/store/appStore'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import type { SectionDTO } from './mockedData'
+
+type GovernanceWithNavigationProps = {
+  dataspaceId: string
+  sections: SectionDTO[]
+  sectionId?: string
+  subsectionId?: string
+}
+
+export const GovernanceNavigation: FC<GovernanceWithNavigationProps> = ({
+  dataspaceId,
+  sections,
+  sectionId,
+  subsectionId
+}) => {
+  const theme = useAppStore(state => state.theme)
+
+  const firstSectionId = sections[0].id
+  const firstSubsectionId = sections[0].contains[0].id
+
+  const [activeSectionId, setActiveSectionId] = useState<string>(sectionId ?? firstSectionId)
+  const [activeSubsectionId, setActiveSubsectionId] = useState<string>(
+    subsectionId ?? firstSubsectionId
+  )
+  useEffect(() => {
+    setActiveSectionId(sectionId ?? firstSectionId)
+    setActiveSubsectionId(subsectionId ?? firstSubsectionId)
+  }, [sectionId, subsectionId, firstSectionId, firstSubsectionId])
+
+  const handleNavSectionClick = useCallback(
+    (section: SectionDTO) => () => {
+      setActiveSectionId(section.id)
+      setActiveSubsectionId(section.contains[0].id)
+    },
+    []
+  )
+  const handleNavSubsectionClick = useCallback(
+    (sectionId: string, subsectionId: string) => () => {
+      setActiveSubsectionId(subsectionId)
+    },
+    []
+  )
+
+  const governanceBasePath = `/dataverse/dataspace/${dataspaceId}/governance`
+
+  return (
+    <nav className="okp4-dataverse-portal-governance-page-navigation">
+      <ul className="okp4-dataverse-portal-governance-page-navigation-section-list">
+        {sections.map(section => {
+          const { id: sectionId, title: sectionTitle, contains: subsections } = section
+          const isSectionActive = sectionId === activeSectionId
+          const hasMoreThanOneSubsection = subsections.length > 1
+          return (
+            <li
+              className="okp4-dataverse-portal-governance-page-navigation-section-list-item"
+              key={sectionId}
+            >
+              <NavLink
```

Comment:
Please use `NavLink` auto matching that directly returns the `isActive` className.